android-sdk/src/main/java/org/sagebionetworks/bridge/android/manager/ProxiedForConsentedUsersApi.java

```diff
@@ -31,6 +31,7 @@
 /**
  * Proxies ForConsentedUserApi calls so it uses retrieves authenticated client from AuthenticationManager
  * to make the call.
+ * TODO: push refreshing of credentials into rest-client's authentication handler
  */
 class ProxiedForConsentedUsersApi implements ForConsentedUsersApi {
```

Comment:
Yes, this is odd, At the very least, it belongs in rest-client, as you mentioned. I've added the TODO and created a JIRA to try and point out it's a known issue. 

rest-client handles reauthentication, but ForConsentedUsersApi it offers is bound to an email-password pair. This proxy binds ForConsentedUsersApi to the email-password pair held in AuthenticationManager. With the unproxied client, any class using the API would need to discard instances of ForConsentedUsersApi (or not hold onto to begin with) if the user signs in or out, orpassword changes. An instance of ProxiedForConsentedUsersApi will by synced with email/password called on DataProvider.signIn/signOut.

Certainly, this type of functionality belongs in rest-client. But I think perhaps it's also doing a bit more than it needs to. I could see wanting to be able to reuse the instance for same account (email) across password changes, but needing to reset state or create new objects in the event of a logout/login. 

IMO it's ugly and perhaps does more than it needs, but it does work to keep the Api/AuthMan in sync. I suggest we defer this change for now and figure out the cleanest/best solution when we pick up the JIRA.